src/python/pants/engine/process_test.py

```diff
@@ -36,253 +25,106 @@ def rule_runner() -> RuleRunner:
     return RuleRunner(
         rules=[
             QueryRule(BinaryPaths, [BinaryPathRequest]),
+            QueryRule(ProcessResult, [Process]),
+            QueryRule(FallibleProcessResult, [Process]),
         ],
     )
 
 
-@dataclass(frozen=True)
-class Concatted:
```

Comment:
The below code was all deleted. It's no longer necessary now that we dogfood the engine so much. Too much complexity to be worth keeping.